spec/controllers/api/assignment_repos_controller_spec.rb

```diff
@@ -35,13 +35,34 @@
         expect(json.first["username"]).to eq(user.github_user.login)
       end
 
-      it "returns repo url" do
-        expect(json.first["repoUrl"]).to eq(@assignment_repo.github_repository.html_url)
-      end
-
       it "returns user display name" do
         expect(json.first["displayName"]).to eq(user.github_user.name)
       end
     end
   end
+
+  describe "GET #clone_url", :vcr do
```

Comment:
Can you add tests in here for cases where the user is not authorized to fetch the clone_url?

Reply:
Are you talking about the case where they have an invalid access token for the API? I test that functionality in the API Application Controller tests, but I could add it here too

Reply:
I think an classroom API level test would be good. That way we can explicitly check how the API will behave when the user isn't authorized.

Reply:
When I made the initial PR for the API, I added authorization tests for `API::ApplicationController` which all of the API controllers extend.

Here are the tests: https://github.com/education/classroom/blob/e350989f35b8e47565f5d240c7e28b346cd54f99/spec/controllers/api/application_controller_spec.rb#L40-L63

I think these are enough to test behavior when the User is not authorized. What do you guys think?